app/src/main/java/com/amaze/filemanager/filesystem/Operations.java

```diff
@@ -441,6 +441,74 @@ public static void rename(
 
       private final DataUtils dataUtils = DataUtils.getInstance();
 
+      /**
+       * Determines whether double rename is required based on original and new file name regardless
+       * of the case-sensitivity of the filesystem
```

Comment:
This _only_ depends on the case sensitivity of the filesystem.

Reply:
I change it to a filesystem case-sensitivity check by checking whether both upper case and lower case input name for file exists return true.

Reply:
This new solution seem to be costly solution and calculating this each time is redundant. If a filesystem is case insensitive then it'll always be case insensitive, so you can persist the result of this in shared preferences and use that from next time.

Reply:
There are multiple local filesystems like sdcard storage and internal storage. The types of filesystem need to be stored with the mount points. Rolling back to previous solution seems easier.

Reply:
Sure please revert back to your old solution. Also note that the new code will double rename even when FS is insensitive and new file and old file names aren't similar at all. 
oldFile.getSimpleName().equalsIgnoreCase(newFile.getSimpleName())
              && !oldFile.getSimpleName().equals(newFile.getSimpleName());

Please also note that I've increased the bounty price for this issue to 10USD you can claim once this PR is merged from the linked issue.